databallpy/utils/game_utils.py

```diff
@@ -120,3 +121,157 @@ def create_event_attributes_dataframe(
+    if tracking_data is not None and not tracking_data.empty:
+        _add_starters_from_tracking_data(metadata, tracking_data)
+    elif event_data is not None and not event_data.empty:
+        _add_starters_from_event_data(metadata, event_data)
+
+    return metadata
+
+
+def _add_starters_from_tracking_data(
+    metadata: Metadata, tracking_data: pd.DataFrame
+) -> None:
+    """Add starter information based on tracking data.
+
+    Identifies the first 22 players (11 from each team) that have non-null tracking data
+    in the first frames of the game.
+
+    Args:
+        metadata (Metadata): The metadata object to update
+        tracking_data (pd.DataFrame): The tracking data
+    """
+    # Get the first frame of the first period
+    first_period = metadata.periods_frames[metadata.periods_frames["period_id"] == 1]
+    if first_period.empty or "start_frame" not in first_period.columns:
+        return
+
+    start_frame = first_period["start_frame"].iloc[0]
+
+    # Find the first frame in tracking data
+    first_frame_data = tracking_data[tracking_data["frame"] == start_frame]
+    if first_frame_data.empty:
+        # Use the very first frame available
+        first_frame_data = tracking_data.iloc[[0]]
+
+    # Get all player columns (those ending with _x)
+    player_x_cols = [col for col in tracking_data.columns if col.endswith("_x")]
+
+    # Separate home and away players
+    home_cols = [col for col in player_x_cols if col.startswith("home_")]
+    away_cols = [col for col in player_x_cols if col.startswith("away_")]
+
+    # Find players with non-null data in the first frame
+    home_starters = []
+    away_starters = []
+
+    for col in home_cols:
+        if first_frame_data[col].notna().any():
+            shirt_num = int(col.split("_")[1])
+            home_starters.append(shirt_num)
+
+    for col in away_cols:
+        if first_frame_data[col].notna().any():
+            shirt_num = int(col.split("_")[1])
+            away_starters.append(shirt_num)
+
+    # Update metadata with starter information
+    metadata.home_players["starter"] = metadata.home_players["shirt_num"].isin(
+        home_starters
+    )
+    metadata.away_players["starter"] = metadata.away_players["shirt_num"].isin(
+        away_starters
+    )
+
+
+def _add_starters_from_event_data(metadata: Metadata, event_data: pd.DataFrame) -> None:
+    """Add starter information based on event data.
+
+    Uses substitute events to determine starters. A player is a starter if they
+    performed an event before the first substitute event.
+
+    Args:
+        metadata (Metadata): The metadata object to update
+        event_data (pd.DataFrame): The event data
+    """
+    if event_data.empty or "databallpy_event" not in event_data.columns:
+        return
+
+    # Find substitute events (assuming they are marked in some way)
+    # Look for substitute/substitution related event types
+    substitute_mask = (
+        event_data["event_type"]
```

Comment:
change "event_type" to "original_event" as it is one of the original column names of the EventDate